New Design

Aksh-Bansal-dev opened the pull request:
![Screenshot from 2022-05-29 01-14-17](https://user-images.githubusercontent.com/63552235/170840852-a919fc01-a24b-4ac2-9712-33459703174c.png)
![Screenshot from 2022-05-29 01-14-31](https://user-images.githubusercontent.com/63552235/170840854-c7f98d7d-b49e-4940-9237-0a481f0dfe98.png)
![Screenshot from 2022-05-29 01-14-41](https://user-images.githubusercontent.com/63552235/170840857-21d7ce42-9009-4c27-9ef9-7a19cc22d102.png)

Aksh-Bansal-dev committed:
feat(frontend): new homepage design

Aksh-Bansal-dev committed:
feat(frontend): new domains page design

Aksh-Bansal-dev committed:
feat(frontend): new login/register design

Aksh-Bansal-dev committed:
chore: change image name

Aksh-Bansal-dev committed:
feat: redesign project page

Aksh-Bansal-dev committed:
feat: add footer section on homepage

Added club's email, insta, github and linkedin.

Samy-33 reviewed (changes requested):
LGTM

Just a few points

1. Compress the images using https://compresspng.com/
2. Question: The last image of the description, it's not clear what is shown? If it's a modal then the backdrop should be uniform. I can see white area at the bottom. I don't understand what it is. Can you please clarify?

Samy-33 commented on frontend/src/App.css:
```diff
@@ -12,18 +12,21 @@ body {
   --primary-btn-color: #3e3d55;
   --nav-btn-background: #707070;
-  --background-gradient: linear-gradient(to bottom right, #3650c7, #42abdc);
+  --background-gradient: #00050b;
```
It's not a gradient anymore. Rename the var to reflect the same.

Samy-33 commented on frontend/src/components/Cards/DomainCard_v2.tsx:
```diff
@@ -0,0 +1,114 @@
+import React from "react";
```
Instead of naming the component/file, create another folder named v2 and include this file in that folder.

Path of this component would then be: `frontend/src/components/Cards/v2/DomainCard.tsx`

Do this for all the renamed components/files

Aksh-Bansal-dev replied:
Maybe I should just delete the older components and rename the newer ones.

Like: `frontend/src/components/Cards/DomainCard_v2.tsx` --> `frontend/src/components/Cards/DomainCard.tsx`

Samy-33 replied:
That makes sense as well.

Samy-33 commented on frontend/src/components/Cards/DomainCard_v2.tsx:
```diff
@@ -0,0 +1,114 @@
+      border: "2px solid white",
+      transform: "translateY(-6vh)",
+    },
+    icoCon: {
```
This name is confusing. Rename it to reflect what it's used for.

Samy-33 commented on frontend/src/components/Cards/DomainCard_v2.tsx:
```diff
@@ -0,0 +1,114 @@
+  github,
+  email,
+  profileImg,
+  linkedIn,
```
Rename these variables so that their types are obvious as well. For example: `github` can be renamed to `bitbyteGithubUrl`

Samy-33 commented on frontend/src/components/Cards/ProjectCardv3.tsx:
```diff
@@ -0,0 +1,98 @@
+  projectName: string;
+  lang: string;
+  description: string;
+  dev: string;
```
Purpose of this variable not clear from the name.

Samy-33 commented on frontend/src/components/Cards/ProjectCardv3.tsx:
```diff
@@ -0,0 +1,98 @@
+  image,
+}) => {
+  const classes = useStyles();
+  const [display, setDisplay] = React.useState<string>("none");
```
Instead of storing `display` in the state, store a boolean `showInfo`

```ts
const [showInfo, setShowInfo] = useState<boolean>(false);
```

then set display style on the basis of this variable. This will separate the state data from CSS styles.

Samy-33 commented on frontend/src/components/Cards/ProjectCardv3.tsx:
```diff
@@ -0,0 +1,98 @@
+      fontSize: "1rem",
+      opacity: 0.5,
+      fontFamily: "monospace",
+    },
```
Use consistent style for classNames and the variables they wrap:

For example, here we see `body` as the className but the variable this class wraps is called: `description`
They could be the same. Goal is to reduce the cognitive load on the reader.

Samy-33 commented on frontend/src/components/FormikTextField.tsx:
```diff
@@ -16,6 +17,7 @@ const FormikTextField: React.FC<FormikTextFieldProps> = ({
   ...props
 }) => {
   // eslint-disable-next-line @typescript-eslint/ban-types
+  const classes = useSignupLoginStyles();
```
I don't think this is a good way. We are using `SignupLoginStyles` for a generic `TextField` component. It would be better to define it's styles. This would remove dependency between them.

Samy-33 commented on frontend/src/components/HomeFooter/index.tsx:
```diff
@@ -0,0 +1,27 @@
+const HomeFooter: React.FC = () => {
+  return (
+    <div className={styles.container}>
+      <a href="https://www.instagram.com/bitbyte.tpc/" target="_blank">
```
These urls and emails should go in a toplevel config file.

Samy-33 commented on frontend/src/components/Navs/Nav_v2/index.tsx:
```diff
@@ -0,0 +1,135 @@
+const Nav2: React.FC = () => {
+  const token = useTokenStore((state) => state.token);
+  const setToken = useTokenStore((state) => state.setToken);
+  const classes = useStyles();
```
Rename to `styles`

Aksh-Bansal-dev replied:
I think renaming `classes` to `styles` would be unnecessary as the entire app uses the `classes` keyword for material-ui styles hook.

Samy-33 replied:
I might be wrong. At some places I did see `styles` being used. My point is to be consistent with terminology.

We want code to be as easy to read as possible. This might sound like an overkill but we can't reach a place if we don't even walk in that direction.

Aksh-Bansal-dev replied:
I used `styles` wherever(very few components) I used CSS modules for styling. So, maybe I should change them to `classes`.

Samy-33 commented on frontend/src/components/Navs/Nav_v2/index.tsx:
```diff
@@ -0,0 +1,135 @@
+  const token = useTokenStore((state) => state.token);
+  const setToken = useTokenStore((state) => state.setToken);
+  const classes = useStyles();
+  const [drawerOpen, setdrawerOpen] = React.useState(false);
```
setdrawerOpen -> setDrawerOpen

Samy-33 commented on frontend/src/components/Navs/Nav_v2/index.tsx:
```diff
@@ -0,0 +1,135 @@
+
+  const toggleDrawer = () => {
+    setdrawerOpen(!drawerOpen);
+    return undefined;
```
Unnecessary return?

Samy-33 commented on frontend/src/components/Navs/Nav_v2/index.tsx:
```diff
@@ -0,0 +1,135 @@
+    }
+  };
+
+  const drawerList = () => (
```
Is there a reason for it to be a function?

Samy-33 commented on frontend/src/components/Navs/Nav_v2/index.tsx:
```diff
@@ -0,0 +1,135 @@
+      <List>
+        {linklist.map((e, key) => {
+          if (e.loginReq) {
+            if (!!token)
```
This could be converted to a `?:` conditional

```tsx
return !!token
  ? <div>...</div>
  : null;
```

Samy-33 commented on frontend/src/components/Navs/Nav_v2/index.tsx:
```diff
@@ -0,0 +1,135 @@
+      onClick={toggleDrawer}
+    >
+      <List>
+        {linklist.map((e, key) => {
```
`e` could be more descriptive like: `linkData`

Samy-33 commented on frontend/src/components/WhyUs/style.module.css:
```diff
@@ -0,0 +1,77 @@
+  grid-template-areas: "content vectors";
+}
+.content {
+  grid-area: content;
```
We are using two layout systems: `flex` and `grid`. We should be consistent and use just one. Create an issue to refactor code to use just one.

Aksh-Bansal-dev replied:
I don't think using flexbox and grid in the same app is an issue. They both have some similarities but both are good for different use cases.

I agree that this component can be built using flexbox but I don't think it's worth changing it.

Samy-33 commented on frontend/src/components/aboutUs/index.tsx:
```diff
@@ -3,63 +3,53 @@ import styles from "./style.module.css";
+      <div className={styles.vectors}>
+        <img
+          className={styles.aboutImg}
+          src="/webix.iiitdmj.ac.in/images/about/btn_7.png"
```
Is it possible to rename these images to something more descriptive?

Samy-33 commented on frontend/src/pages/Domains/domainList.ts:
```diff
@@ -0,0 +1,37 @@
+export const domainList = [
```
Add a comment on top to make it clear that this list contains dummy data and we need to update it. Preferably create an issue and link it here:
```
// todo(link/to/the/issue): update the list to contain correct data.
```

Samy-33 commented on frontend/src/pages/Domains/index.tsx:
```diff
@@ -27,61 +44,69 @@ const useStyles = makeStyles(() =>
+          We are everywhere
+        </Typography>
+        <Typography className={classes.subHeading} align="center">
+          {"<p>From web to app, AI to ML, Blockchain to DevOps</p>"}
```
Why is this wrapped within `{}`?

Aksh-Bansal-dev replied:
We want to display `<p>From web to app, AI to ML, Blockchain to DevOps</p>`, so I used `{}` to treat `<p></p>` as a string instead of HTML tag.

Samy-33 replied:
Got it.

Samy-33 commented on frontend/src/pages/Domains/index.tsx:
```diff
@@ -27,61 +44,69 @@ const useStyles = makeStyles(() =>
         </Typography>
-        <Container className={classes.flexbox}>
+        <div className={classes.flexbox}>
           {teamlist.map((e, i) => {
```
Descriptive names and camelCase:

```ts
teamList.map((team, index) => {...}
```

Samy-33 commented on frontend/src/pages/Domains/index.tsx:
```diff
@@ -27,61 +44,69 @@ const useStyles = makeStyles(() =>
+        <div className={classes.flexbox}>
           {teamlist.map((e, i) => {
             if (e.team === "coordinator") {
-              return <DomainCard key={i} {...e} />;
+              return <DomainCard2 key={i} {...{ ...e, team: "Coordinator" }} />;
+            } else {
+              return <></>;
```
Could be converted to `?:` conditional.

Aksh-Bansal-dev commented:
> 2. Question: The last image of the description, it's not clear what is shown? If it's a modal then the backdrop should be uniform. I can see white area at the bottom. I don't understand what it is. Can you please clarify?


It's the projects page. The card displays additional info on hover and just the image otherwise.

![Screenshot from 2022-05-29 11-47-57](https://user-images.githubusercontent.com/63552235/170854999-67ab0993-e13a-4529-bbb8-1db2c04d9175.png)

Aksh-Bansal-dev committed:
chore: compress images

Aksh-Bansal-dev committed:
refactor(frontend): improve code quality

- Changed variable names to improve readability.
- Separated styles in `signupLoginStyle.ts` to make code more
maintainable.
- Moved club's social media urls and email to global.

Aksh-Bansal-dev committed:
chore: remove unused code from prev design

Aksh-Bansal-dev committed:
feat(frontend): improve homepage css

- add new font
- increase font size
- improve margin/padding
- replace png with svg for better quality

Samy-33 reviewed (approved):
LGTM
